rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1750,10 +1702,9 @@ extern "C"
             {
                 void *data = guard_conditions->guard_conditions[i];
                 GuardCondition *guard_condition = (GuardCondition*)data;
-                if(!guard_condition->getHasTriggered())
-                {
-                    guard_conditions->guard_conditions[i] = 0;
-                }
+                // Guard conditions persist (don't set to null)
+                // We call getHasTriggered to reset the trigger state.
```

Comment:
> By not having a way to detect which conditions triggered the caller must always "handle" all of them. If in the event of a condition the processing is significant the caller has no way of doing that overhead conditionally. Therefore I think it makes sense to keep the ability to signal the triggered conditions.

I think @jacquelinekay's point is that this is another example of a solution in search of a problem. Without being able to point to a scenario where we need to do this it might not be worth keeping. Another way to look at it is that if we're not using it (even just in a test) then it's very likely it will be broken at some point and no one will notice.